From 8f1495c2f4a0eab2bbcb6c99d4946e0bbe23fc0f Mon Sep 17 00:00:00 2001
From: Changhyeok Bae <changhyeok.bae@lge.com>
Date: Sat, 5 Nov 2016 17:08:37 +0900
Subject: [PATCH] Fix broken resolution issue

---
 app/qml-example/main.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/app/qml-example/main.cpp b/app/qml-example/main.cpp
index a60213b..2cf37bd 100644
--- a/app/qml-example/main.cpp
+++ b/app/qml-example/main.cpp
@@ -42,6 +42,9 @@ int main(int argc, char *argv[])
     QQuickView view(QUrl(QStringLiteral("qrc:/main.qml")));
     object = (QObject *)view.rootObject();
 
+    view.setFlags(Qt::FramelessWindowHint);
+    view.setColor(Qt::transparent);
+
     MyClass myClass;
     QObject::connect(object, SIGNAL(qmlSignal(QString)),
                      &myClass, SLOT(cppSlot(QString)));
-- 
2.9.0

