diff --git a/src/hmi/qml/hmi-launcher/CMakeLists.txt b/src/hmi/qml/hmi-launcher/CMakeLists.txt
index 70cf21d..e84c4e7 100644
--- a/src/hmi/qml/hmi-launcher/CMakeLists.txt
+++ b/src/hmi/qml/hmi-launcher/CMakeLists.txt
@@ -112,8 +112,8 @@ endif()
 # set the paths into the resource.js file
 set(FILE_RESOURCE_IN ${CMAKE_CURRENT_SOURCE_DIR}/../Core/resource.js.in)
 set(FILE_RESOURCE_OUT ${CMAKE_CURRENT_SOURCE_DIR}/../Core/resource.js)
-set(CONSTANTS_PATH ${CMAKE_CURRENT_BINARY_DIR})
-set(ICON_PATH ${NAVIT_PNG_BIN_DIR})
+set(CONSTANTS_PATH .)
+set(ICON_PATH /usr/share/navit/xpm)
 configure_file(${FILE_RESOURCE_IN} ${FILE_RESOURCE_OUT} @ONLY)
 
