Index: git/src/genivilogreplayer/CMakeLists.txt
===================================================================
--- git.orig/src/genivilogreplayer/CMakeLists.txt
+++ git/src/genivilogreplayer/CMakeLists.txt
@@ -20,12 +20,14 @@ cmake_minimum_required(VERSION 2.8)
 include(FindPkgConfig)
 pkg_search_module(GLIB glib-2.0)
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+
 set(genivilogreplayer_headers genivilogreplayerplugin.h)
 set(genivilogreplayer_sources genivilogreplayerplugin.cpp)
 
 add_library(genivilogreplayerplugin MODULE ${genivilogreplayer_sources})
 set_target_properties(genivilogreplayerplugin PROPERTIES PREFIX "")
 include_directories(${AUTOMOTIVE_MESSAGE_BROKER_SRC_DIR}/lib ${GLIB_INCLUDE_DIRS})
-target_link_libraries(genivilogreplayerplugin amb -L${AUTOMOTIVE_MESSAGE_BROKER_LIB_DIR})
+target_link_libraries(genivilogreplayerplugin amb)

 install(TARGETS genivilogreplayerplugin LIBRARY DESTINATION lib/automotive-message-broker)
