From e260a77176040fe3edc85f3d25c335272c9844d8 Mon Sep 17 00:00:00 2001
From: Changhyeok Bae <changhyeok.bae@lge.com>
Date: Mon, 4 Jul 2016 13:38:54 +0900
Subject: [PATCH] Fix broken resolution issue

---
 app/gdp-hmi-launcher2/main.cpp  | 1 -
 1 files changed, 1 deletions(-)

diff --git a/app/gdp-hmi-launcher2/main.cpp b/app/gdp-hmi-launcher2/main.cpp
index 8a399bd..00613e0 100755
--- a/app/gdp-hmi-launcher2/main.cpp
+++ b/app/gdp-hmi-launcher2/main.cpp
@@ -39,7 +39,6 @@ int main(int argc, char* argv[])
 #endif
 
     setenv("QT_QPA_PLATFORM", "wayland", 1); // force to use wayland plugin
-    setenv("QT_WAYLAND_DISABLE_WINDOWDECORATION", "1", 1);
 
 #ifdef USE_DLT
     DLT_LOG(launcherTraceCtx,DLT_LOG_INFO,DLT_STRING("GDP: launcher2 surface id="),DLT_INT(GDP_LAUNCHER2_SURFACE_ID));
-- 
1.9.1

