Index: git/src/navigation/map-viewer/CMakeLists.txt
===================================================================
--- git.orig/src/navigation/map-viewer/CMakeLists.txt
+++ git/src/navigation/map-viewer/CMakeLists.txt
@@ -25,15 +25,15 @@ find_program(XSLTPROC xsltproc REQUIRED)
 # Generate the config file for navit/mapviewer
 set(MAP_VIEWER_SRC ${CMAKE_CURRENT_SOURCE_DIR})
 
-if(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/../navit/navit/navit_genivi_mapviewer.xml)
+if(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/navit_genivi_mapviewer.xml)
 	execute_process(
-		OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/../navit/navit/navit_genivi_mapviewer.xml
+		OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/navit_genivi_mapviewer.xml
 		COMMAND ${XSLTPROC} ${MAP_VIEWER_SRC}/navit_genivi_mapviewer.xsl ${navit_SRC_DIR}/navit/navit/navit_shipped.xml        
 	)
 else()
-	if(${MAP_VIEWER_SRC}/navit_genivi_mapviewer.xsl IS_NEWER_THAN ${CMAKE_CURRENT_BINARY_DIR}/../navit/navit/navit_genivi_mapviewer.xml)
+	if(${MAP_VIEWER_SRC}/navit_genivi_mapviewer.xsl IS_NEWER_THAN ${CMAKE_CURRENT_BINARY_DIR}/navit_genivi_mapviewer.xml)
 		execute_process(
-			OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/../navit/navit/navit_genivi_mapviewer.xml
+			OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/navit_genivi_mapviewer.xml
 			COMMAND ${XSLTPROC} ${MAP_VIEWER_SRC}/navit_genivi_mapviewer.xsl ${navit_SRC_DIR}/navit/navit/navit_shipped.xml        
 		)
 	endif()
Index: git/src/navigation/navigation-core/CMakeLists.txt
===================================================================
--- git.orig/src/navigation/navigation-core/CMakeLists.txt
+++ git/src/navigation/navigation-core/CMakeLists.txt
@@ -25,15 +25,15 @@ find_program(XSLTPROC xsltproc REQUIRED)
 # Generate the config file for navit/navigation-core
 set(NAVIGATION_CORE_SRC ${CMAKE_CURRENT_SOURCE_DIR})
 
-if(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/../navit/navit/navit_genivi_navigationcore.xml)
+if(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/navit_genivi_navigationcore.xml)
 	execute_process(
-		OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/../navit/navit/navit_genivi_navigationcore.xml
+		OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/navit_genivi_navigationcore.xml
 		COMMAND ${XSLTPROC} ${NAVIGATION_CORE_SRC}/navit_genivi_navigationcore.xsl ${navit_SRC_DIR}/navit/navit/navit_shipped.xml        
 	)
 else()
-	if(${NAVIGATION_CORE_SRC}/navit_genivi_navigationcore.xsl IS_NEWER_THAN ${CMAKE_CURRENT_BINARY_DIR}/../navit/navit/navit_genivi_navigationcore.xml)
+	if(${NAVIGATION_CORE_SRC}/navit_genivi_navigationcore.xsl IS_NEWER_THAN ${CMAKE_CURRENT_BINARY_DIR}/navit_genivi_navigationcore.xml)
 		execute_process(
-			OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/../navit/navit/navit_genivi_navigationcore.xml
+			OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/navit_genivi_navigationcore.xml
 			COMMAND ${XSLTPROC} ${NAVIGATION_CORE_SRC}/navit_genivi_navigationcore.xsl ${navit_SRC_DIR}/navit/navit/navit_shipped.xml        
 		)
 	endif()
