From 7dd2c6e2a7c375d2fb259933af20ad688740a463 Mon Sep 17 00:00:00 2001
From: Changhyeok Bae <changhyeok.bae@gmail.com>
Date: Mon, 19 Jun 2017 02:09:52 +0900
Subject: [PATCH] Temporarily code block

---
 AudioManagerCore/src/CAmDatabaseHandlerMap.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/AudioManagerCore/src/CAmDatabaseHandlerMap.cpp b/AudioManagerCore/src/CAmDatabaseHandlerMap.cpp
index 48ef37b..2337c2c 100644
--- a/AudioManagerCore/src/CAmDatabaseHandlerMap.cpp
+++ b/AudioManagerCore/src/CAmDatabaseHandlerMap.cpp
@@ -655,11 +655,6 @@ am_Error_e CAmDatabaseHandlerMap::enterSinkDB(const am_Sink_s & sinkData, am_sin
     	logError(__METHOD_NAME__,"sinkName must not be zero");
     	return (E_NOT_POSSIBLE);
     }
-    if(!existSinkClass(sinkData.sinkClassID))
-    {
-    	logError(__METHOD_NAME__,"sinkClass must be valid");
-    	return (E_NOT_POSSIBLE);
-    }
 
     if(!(sinkData.muteState>=MS_UNKNOWN && sinkData.muteState<=MS_MAX))
     {
-- 
1.9.1

