From eb42545e66d05fce084d9ec74c8fbd926a5f74f0 Mon Sep 17 00:00:00 2001
From: Changhyeok Bae <changhyeok.bae@gmail.com>
Date: Thu, 22 Jun 2017 15:35:15 +0900
Subject: [PATCH 1/2] Correct label on RPI

Signed-off-by: a629114 <a629114@aw.atos.net>
---
 src/controls/Label.qml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/controls/Label.qml b/src/controls/Label.qml
index 7174b28..121c3c1 100644
--- a/src/controls/Label.qml
+++ b/src/controls/Label.qml
@@ -83,7 +83,6 @@ Text {
     id: label
     color: SystemPaletteSingleton.windowText(enabled)
     activeFocusOnTab: false
-    renderType: Settings.isMobile ? Text.QtRendering : Text.NativeRendering
     Accessible.name: text
     Accessible.role: Accessible.StaticText
 }
-- 
1.9.1

