From 9cf1db27878a02307febce96ff23487a15d30ecb Mon Sep 17 00:00:00 2001
From: a629114 <a629114@aw.atos.net>
Date: Thu, 1 Sep 2016 16:34:31 +0200
Subject: [PATCH] Correct label on RPI

---
 src/controls/Label.qml | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/controls/Label.qml b/src/controls/Label.qml
index b07e96d..9835df3 100644
--- a/src/controls/Label.qml
+++ b/src/controls/Label.qml
@@ -82,8 +82,7 @@ Text {
 
     id: label
     color: pal.windowText
-    activeFocusOnTab: false
-    renderType: Settings.isMobile ? Text.QtRendering : Text.NativeRendering
+    activeFocusOnTab: false    
     SystemPalette {
         id: pal
         colorGroup: enabled ? SystemPalette.Active : SystemPalette.Disabled
-- 
2.1.4

