From c4feac3e1914f4e9e591a2cb00e95ad1d8f6ca07 Mon Sep 17 00:00:00 2001
From: Changhyeok Bae <changhyeok.bae@gmail.com>
Date: Thu, 22 Jun 2017 15:37:57 +0900
Subject: [PATCH 2/2] Correct textfield on RPI

Signed-off-by: a629114 <a629114@aw.atos.net>
---
 src/controls/TextField.qml | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/controls/TextField.qml b/src/controls/TextField.qml
index c5d5d2b..fbe5581 100644
--- a/src/controls/TextField.qml
+++ b/src/controls/TextField.qml
@@ -629,7 +629,6 @@ Control {
         color: __panel ? __panel.placeholderTextColor : "darkgray"
         clip: contentWidth > width;
         elide: Text.ElideRight
-        renderType: __style ? __style.renderType : Text.NativeRendering
     }
 
     TextInputWithHandles {
@@ -656,8 +655,6 @@ Control {
         color: __panel ? __panel.textColor : "darkgray"
         clip: contentWidth > width
 
-        renderType: __style ? __style.renderType : Text.NativeRendering
-
         Keys.forwardTo: textfield
 
         EnterKey.type: control.EnterKey.type
-- 
1.9.1

