From e260a77176040fe3edc85f3d25c335272c9844d8 Mon Sep 17 00:00:00 2001
From: Changhyeok Bae <changhyeok.bae@lge.com>
Date: Mon, 4 Jul 2016 13:38:54 +0900
Subject: [PATCH] Fix broken resolution issue

---
 app/gdp-hmi-background/main.cpp | 3 +++
 1 files changed, 3 insertions(+)

diff --git a/app/gdp-hmi-background/main.cpp b/app/gdp-hmi-background/main.cpp
index 42b433c..fe886b1 100644
--- a/app/gdp-hmi-background/main.cpp
+++ b/app/gdp-hmi-background/main.cpp
@@ -32,6 +32,9 @@ int main(int argc, char *argv[])
     QGuiApplication app(argc, argv);
     QQuickView view(QUrl(QStringLiteral("qrc:/main.qml")));
 
+    view.setFlags(Qt::FramelessWindowHint);
+    view.setColor(Qt::transparent);
+
     view.setProperty("IVI-Surface-ID", GDP_BACKGROUND_SURFACE_ID);
     view.show();
 
-- 
1.9.1

