Index: git/demoui/main.cpp
===================================================================
--- git.orig/demoui/main.cpp
+++ git/demoui/main.cpp
@@ -20,6 +20,8 @@
 #include <QtQml>
 #include <QQuickItem>
 
+#define BROWSER_POC_SURFACE_ID 30
+
 int main(int argc, char *argv[])
 {
     QGuiApplication app(argc, argv);
@@ -29,6 +31,7 @@ int main(int argc, char *argv[])
     qmlRegisterType<BrowserDbus>("browserdbusinterface",1,0,"BrowserInterface");
     qmlRegisterType<Bookmark>("browserdbusinterface",1,0,"Tmp");
 
+    viewer->setProperty("IVI-Surface-ID", BROWSER_POC_SURFACE_ID);
     viewer->setMainQmlFile(QStringLiteral("qml/demoui/main.qml"));
     viewer->setFlags(Qt::WindowStaysOnTopHint|Qt::FramelessWindowHint);
     viewer->setGeometry(0, 0, 1024, 80);
