From 6cf2b01cb7475e06f44fb5db6352156f7384aa5b Mon Sep 17 00:00:00 2001
From: Tom Pollard <tom.pollard@codethink.co.uk>
Date: Tue, 1 Dec 2015 14:01:32 +0000
Subject: [PATCH 1/1] libpers_common.so extension fix for EGL

---
 config/pclCustomLibConfigFile.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/pclCustomLibConfigFile.cfg b/config/pclCustomLibConfigFile.cfg
index c9667d1..10183f6 100644
--- a/config/pclCustomLibConfigFile.cfg
+++ b/config/pclCustomLibConfigFile.cfg
@@ -1,7 +1,7 @@
 hwinfo /usr/lib/libhwinfoperscustom.so init async
 secure /usr/lib/libsecureperscustom.so init sync
 custom3 /usr/lib/libcustom3perscustom.so od sync
-default /usr/lib/libpers_common.so.1 init sync
+default /usr/lib/libpers_common.so.0 init sync
 emergency /usr/lib/libemergencyperscustom.so od async
 early /usr/lib/libearlyperscustom.so od sync
 custom2 /usr/lib/libcustom2perscustom.so init sync
-- 
2.1.4

