From d6357d4bca0857521d762b4beef476c253190c31 Mon Sep 17 00:00:00 2001
From: Tom Pollard <tom.pollard@codethink.co.uk>
Date: Tue, 12 Jan 2016 16:34:57 +0000
Subject: [PATCH 1/1] qwaylandwindow: add support for IVI-Surface-ID property

---
 src/client/qwaylandwindow.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/client/qwaylandwindow.cpp b/src/client/qwaylandwindow.cpp
index 12d2089..22fd1e1 100644
--- a/src/client/qwaylandwindow.cpp
+++ b/src/client/qwaylandwindow.cpp
@@ -105,7 +105,14 @@ QWaylandWindow::QWaylandWindow(QWindow *window)
 
     if (!(window->flags() & Qt::BypassWindowManagerHint)) {
         if (mDisplay->shellIvi()) {
-            mShellSurface = new QWaylandIviSurface(mDisplay->shellIvi()->surface_create(IVI_SURFACE_ID + getpid(), object()), this);
+                unsigned int id = 0;
+                QVariant value = window->property("IVI-Surface-ID");
+                if (value.isValid()) {
+                    id = value.toUInt();
+                } else {
+                    id = IVI_SURFACE_ID + getpid();
+                }
+                mShellSurface = new QWaylandIviSurface(mDisplay->shellIvi()->surface_create(id, object()), this);
         } else
             mShellSurface = mDisplay->createShellSurface(this);
     }
-- 
1.9.1

