diff --git a/src/client/qwaylandinputdevice.cpp b/src/client/qwaylandinputdevice.cpp
index 08e2945..37cb66a 100644
--- a/src/client/qwaylandinputdevice.cpp
+++ b/src/client/qwaylandinputdevice.cpp
@@ -836,6 +836,7 @@ void QWaylandInputDevice::Touch::touch_down(uint32_t serial,
     Q_UNUSED(time);
     mFocus = QWaylandWindow::fromWlSurface(surface);
     mParent->handleTouchPoint(id, wl_fixed_to_double(x), wl_fixed_to_double(y), Qt::TouchPointPressed);
+    touch_frame();
 }
 
 void QWaylandInputDevice::Touch::touch_up(uint32_t serial, uint32_t time, int32_t id)
@@ -844,12 +845,14 @@ void QWaylandInputDevice::Touch::touch_up(uint32_t serial, uint32_t time, int32_
     Q_UNUSED(time);
     mFocus = 0;
     mParent->handleTouchPoint(id, 0, 0, Qt::TouchPointReleased);
+    touch_frame();
 }
 
 void QWaylandInputDevice::Touch::touch_motion(uint32_t time, int32_t id, wl_fixed_t x, wl_fixed_t y)
 {
     Q_UNUSED(time);
     mParent->handleTouchPoint(id, wl_fixed_to_double(x), wl_fixed_to_double(y), Qt::TouchPointMoved);
+    touch_frame();
 }
 
 void QWaylandInputDevice::Touch::touch_cancel()
