From 012f9cb91006d5507d4048cd121cd78115f1d300 Mon Sep 17 00:00:00 2001
From: Changhyeok Bae <changhyeok.bae@lge.com>
Date: Mon, 25 Jul 2016 15:54:03 +0900
Subject: [PATCH] Fix issue not to apply pam definition

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index b89608d..ca25c9a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -418,7 +418,7 @@ AS_IF([test "x$enable_resize_optimization" = "xyes"],
 
 AC_ARG_ENABLE(weston-launch, [  --enable-weston-launch],, enable_weston_launch=yes)
 AM_CONDITIONAL(BUILD_WESTON_LAUNCH, test x$enable_weston_launch == xyes)
-if test x$enable_weston_launch == xyes -a x$use_pam = xyes; then
+if test x$enable_weston_launch == xyes -a x$with_pam = xyes; then
   WESTON_SEARCH_LIBS([PAM], [pam], [pam_open_session], [have_pam=yes], [have_pam=no])
   if test x$have_pam == xno; then
     AC_ERROR([PAM support is explicitly requested, but libpam couldn't be found])
@@ -677,7 +677,7 @@ AC_MSG_RESULT([
 	Enable developer documentation	${enable_devdocs}
 
 	weston-launch utility		${enable_weston_launch}
-	PAM support			${use_pam}
+	PAM support			${with_pam}
 	systemd-login support		${have_systemd_login}
 	systemd notify support		${enable_systemd_notify}
 
-- 
1.9.1

