Index: git/src/hmi/qml/hmi-launcher/CMakeLists.txt
===================================================================
--- git.orig/src/hmi/qml/hmi-launcher/CMakeLists.txt
+++ git/src/hmi/qml/hmi-launcher/CMakeLists.txt
@@ -69,7 +69,7 @@ set(CMAKE_AUTOMOC ON)
 add_definitions(${QT5_CORE_DEFINITIONS})
 add_definitions(-DQT_PLUGIN)
 add_definitions(-DQT_NO_DEBUG)
-set(qt_moc moc -o)
+set(qt_moc ${QT_MOC} -o)
 set(MOC moc)
 set(MOC_DIR ${CMAKE_CURRENT_BINARY_DIR}/${MOC})
 
